tests/phpunit/CRM/MembersOnlyEvent/Service/MembersOnlyEventAccessTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+use CRM_MembersOnlyEvent_Service_MembersOnlyEventAccess as MembersOnlyEventAccess;
+use CRM_MembersOnlyEvent_Test_Fabricator_MembersOnlyEvent as MembersOnlyEventFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_Contact as ContactFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_Membership as MembershipFabricator;
+use CRM_MembersOnlyEvent_BAO_EventMembershipType as EventMembershipType;
+
+require_once __DIR__ . '/../../../BaseHeadlessTest.php';
+
+/**
+ * Class CRM_MembersOnlyEvent_BAO__MembersOnlyEventPriceFieldValue
+ *
+ * @group headless
+ */
+class CRM_MembersOnlyEvent_Service_MembersOnlyEventAccessTest extends BaseHeadlessTest {
+
+  /**
+   * @param int $value
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function setMembershipDurationCheckSettings($value) {
+    civicrm_api3('Setting', 'create', [
+      'membership_duration_check' => $value,
+    ]);
+  }
+
+  /**
+   * @return int
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function createLoggedInUser() {
+    $contactID = ContactFabricator::fabricate()['id'];
+
+    $session = CRM_Core_Session::singleton();
+    $session->set('userID', $contactID);
+    return $contactID;
+  }
+
+  /**
+   * Tests IsValidPath() happy path.
+   */
+  public function testIsValidPathHappyPath() {
+    $config = CRM_Core_Config::singleton();
+    $tmpGlobals = [];
+    $tmpGlobals['_GET'][$config->userFrameworkURLVar] = 'civicrm/event/info';
+    CRM_Utils_GlobalStack::singleton()->push($tmpGlobals);
+
+    $membersOnlyEventAccess = new MembersOnlyEventAccess();
+    $this->assertTrue($membersOnlyEventAccess->isValidPath());
+
+    CRM_Utils_GlobalStack::singleton()->pop();
+
+    $tmpGlobals['_GET'][$config->userFrameworkURLVar] = 'civicrm/event/register';
+
+    CRM_Utils_GlobalStack::singleton()->push($tmpGlobals);
+
+    $membersOnlyEventAccess = new MembersOnlyEventAccess();
+    $this->assertTrue($membersOnlyEventAccess->isValidPath());
+
+    CRM_Utils_GlobalStack::singleton()->pop();
+  }
+
+  /**
+   * Tests IsValidPath() unhappy path.
+   */
+  public function testIsValidPathUnhappyPath() {
+    $config = CRM_Core_Config::singleton();
+    $tmpGlobals = [];
+    $tmpGlobals['_GET'][$config->userFrameworkURLVar] = 'civicrm/home';
+    CRM_Utils_GlobalStack::singleton()->push($tmpGlobals);
+
+    $membersOnlyEventAccess = new MembersOnlyEventAccess();
+    $this->assertFalse($membersOnlyEventAccess->isValidPath());
+
+    CRM_Utils_GlobalStack::singleton()->pop();
+  }
+
+  /**
+   * Tests redirectUsersWithoutEventAccess().
+   */
+  public function testRedirectUsersWithoutEventAccess() {
+    $membersOnlyEventAccess = new MembersOnlyEventAccess();
+    $this->assertFalse($membersOnlyEventAccess->userHasEventAccess());
```

Comment:
you are testing the redirect here so no need for this assert, same applies to other tests below.. it is generally a good practice to try to have single assert in a single test whenever it's possible, this allows you to separate your tests so each one test very specific functionality and when it fails you know which one exactly failed and for which reason.